src/iop/cacorrect.c

```diff
@@ -340,93 +343,93 @@ void process(
 
   float blockave[2][2] = { { 0, 0 }, { 0, 0 } };
   float blocksqave[2][2] = { { 0, 0 }, { 0, 0 } };
   float blockdenom[2][2] = { { 0, 0 }, { 0, 0 } };
   float blockvar[2][2];
   // order of 2d polynomial fit (polyord), and numpar=polyord^2
-  int polyord = 4, numpar = 16;
+  int polyord = 4;
+  int numpar = 16;
 
-  const float eps = 1e-5f, eps2 = 1e-10f; // tolerance to avoid dividing by zero
+  const float eps = 1e-5f;
+  const float eps2 = 1e-10f; // tolerance to avoid dividing by zero
 
   for(size_t it = 0; it < iterations && processpasstwo; it++)
   {
 
-#if defined(_OPENMP) && !defined(__APPLE__)
-#pragma omp parallel default(none) \
-  dt_omp_firstprivate(in, out, height, width, filters, processpasstwo, hblsz, vblsz, \
-                      blockwt, blockshifts, blockvar, Gtmp, RawDataTmp, \
-                      border, border2, eps, eps2, darktable) \
-  shared(numpar, polyord, fitparams, blockdenom, blockave, blocksqave)
-  // if any of the above shared are made firstprivate, we get the wrong answer
+#if defined(_OPENMP)
+#pragma omp parallel
 #endif
    {
     // direction of the CA shift in a tile
     int GRBdir[2][3];
 
-    int shifthfloor[3], shiftvfloor[3], shifthceil[3], shiftvceil[3];
+    int shifthfloor[3];
+    int shiftvfloor[3];
+    int shifthceil[3];
+    int shiftvceil[3];
 
     // local quadratic fit to shift data within a tile
     float coeff[2][3][2];
     // measured CA shift parameters for a tile
     float CAshift[2][2];
     // polynomial fit coefficients
     // residual CA shift amount within a plaquette
-    float shifthfrac[3], shiftvfrac[3];
+    float shifthfrac[3];
+    float shiftvfrac[3];
     // per thread data for evaluation of block CA shift variance
-    float blockavethr[2][2] = { { 0, 0 }, { 0, 0 } }, blocksqavethr[2][2] = { { 0, 0 }, { 0, 0 } },
-          blockdenomthr[2][2] = { { 0, 0 }, { 0, 0 } };
+    float blockavethr[2][2] = { { 0, 0 }, { 0, 0 } };
+    float blocksqavethr[2][2] = { { 0, 0 }, { 0, 0 } };
+    float blockdenomthr[2][2] = { { 0, 0 }, { 0, 0 } };
 
     // assign working space
-    const size_t buffersize = sizeof(float) * 3 * ts * ts + 6 * sizeof(float) * ts * tsh + 8 * 64 + 63;
-    char *buffer = (char *)malloc(buffersize);
-    char *data = (char *)(((uintptr_t)buffer + (uintptr_t)63) / 64 * 64);
+    const size_t tilebuf_size = ts * ts + 16;
+    const size_t tilebuf_half_size = ts * tsh + 16;
+
+    const size_t buffersize = 3 * tilebuf_size + 6 * tilebuf_half_size;
+    float *data = dt_alloc_align_float(buffersize);
 
     // shift the beginning of all arrays but the first by 64 bytes to avoid cache miss conflicts on CPUs which
     // have <=4-way associative L1-Cache
 
     // rgb data in a tile
     float *rgb[3];
     rgb[0] = (float(*))data;
-    rgb[1] = (float(*))(data + 1 * sizeof(float) * ts * ts + 1 * 64);
-    rgb[2] = (float(*))(data + 2 * sizeof(float) * ts * ts + 2 * 64);
+    rgb[1] = (float(*))(data + tilebuf_size);
```

Comment:
You can get rid of the explicit casts and sets of parentheses now, which makes this piece of code a bit more readable.

Reply:
rigth again